scripts/gbdt_bench/bench_compiled.py

```
@@ -0,0 +1,68 @@
+    rows = np.fromfile(os.path.join(a.dir, "rows.f32"), dtype="<f4").reshape(-1, nf)
+    refp = os.path.join(a.dir, "preds_rust.f32")
+    if not os.path.exists(refp):
+        sys.exit("run the Rust bench first (writes preds_rust.f32)")
+    ref = np.fromfile(refp, dtype="<f4").astype(np.float64)
+    print(f"model {meta['model_id']}: {meta['n_trees']} trees, {meta['n_nodes']} nodes, {nf} features; rows {rows.shape[0]}")
+    txt = os.path.join(a.dir, "frag_lgbm.txt")
+    try:
+        import lightgbm as lgb
+        b = lgb.Booster(model_file=txt)
+        report("lightgbm", lambda x: b.predict(x, num_threads=1), rows, ref)
+    except Exception as e:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)